sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/Samplers.java

```diff
@@ -215,10 +248,35 @@ public String getDescription() {
 
   @Immutable
   static class ParentBased implements Sampler {
-    private final Sampler delegateSampler;
+    private final Sampler root;
+    private Sampler remoteParentSampled = alwaysOn();
+    private Sampler remoteParentNotSampled = alwaysOff();
+    private Sampler localParentSampled = alwaysOn();
+    private Sampler localParentNotSampled = alwaysOff();
+
```

Comment:
I'd prefer if these were `final` and assigned in the constructor. Then, the single-arg constructor can just pass these values in to the other constructor.

Reply:
agree 👍 . thanks.